File: compras.java
import java.util.ArrayList;
import java.util.Scanner;
public class ListaCompras {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        ArrayList<String> lista = new ArrayList<>();
        System.out.print("Quantos itens você quer adicionar? ");
        int quantidade = Integer.parseInt(scanner.nextLine());

        for (int i = 0; i < quantidade; i++) {
            System.out.print("Digite o item: ");
            String item = scanner.nextLine();
            lista.add(item);
        }

        System.out.println("Lista de compras:");
        for (int i = 0; i < lista.size(); i++) {
            System.out.println((i + 1) + " - " + lista.get(i));
        }
    }
}

File: compras.py
compras = []
quantidade = int(input("Quantos itens você quer adicionar? "))
for i in range(quantidade):
    item = input("Digite o item: ")
    compras.append(item)
print("Lista de compras:")
for i in range(len(compras)):
    print(i + 1, "-", compras[i])
